File: erykandroidcommon/src/main/java/pl/edu/amu/wmi/erykandroidcommon/verify/AbstractEditorResultActivity.java
package pl.edu.amu.wmi.erykandroidcommon.verify;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.MenuRes;
import android.view.Menu;
import android.view.MenuItem;

import pl.edu.amu.wmi.erykandroidcommon.R;
import pl.edu.amu.wmi.erykandroidcommon.base.BaseActivity;
import pl.edu.amu.wmi.erykandroidcommon.util.WindowUtil;


public abstract class AbstractEditorResultActivity extends BaseActivity implements FieldVerifier {

    private FormVerificationManager formVerificationManager;

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        getMenuInflater().inflate(getMenuRes(), menu);
        return true;
    }

    @MenuRes
    abstract int getMenuRes();

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == R.id.action_send) {
            if (formVerificationManager.verify()) {
                setResult(1, putExtraResult(new Intent()));
                finish();
            }
        } else if (item.getItemId() == android.R.id.home) {
            onBackPressed();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    protected abstract Intent putExtraResult(Intent resultIntent);

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        formVerificationManager = new FormVerificationManager(this, this);
        WindowUtil.enableActionBar(this);
        changeNotificationBarColor(this);
    }

}

File: src/main/java/pl/edu/amu/wmi/erykandroidcommon/verify/AbstractEditorResultActivity.kt
package pl.edu.amu.wmi.erykandroidcommon.verify

import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import pl.edu.amu.wmi.erykandroidcommon.R
import pl.edu.amu.wmi.erykandroidcommon.base.BaseActivity
import pl.edu.amu.wmi.erykandroidcommon.helper.WindowUtil

abstract class AbstractEditorResultActivity : BaseActivity(), FieldVerifier {

    private val menuRes = R.menu.post_editor_menu
    private lateinit var manager: FormVerificationManager

    override fun onCreateOptionsMenu(menu: Menu): Boolean {
        menuInflater.inflate(menuRes, menu)
        return true
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == R.id.action_send) {
            if (manager.verify()) {
                setResult(1, putExtraResult(Intent()))
                finish()
            }
        } else if (item.itemId == android.R.id.home) {
            onBackPressed()
            return true
        }
        return super.onOptionsItemSelected(item)
    }

    protected abstract fun putExtraResult(resultIntent: Intent): Intent

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        WindowUtil.enableActionBar(this)
        manager = FormVerificationManager(this, this)
    }
}
